### PR TITLE
assessment: verify contents of .level_group file.

### PR DESCRIPTION
**Pull request by breville**
Specifically: no duplicate levels; only multi and text_match levels.

### CODE REVIEW

File: dashboard/app/dsl/level_group_dsl.rb [hunk 1]
```diff
@@ -27,6 +28,19 @@ def page
...
 
   def level(name)
+    # Ensure level name hasn't already been used.
+    if @level_names.include? name
+      raise "Don't use the same level twice in a LevelGroup (#{name})."
+    end
+    @level_names << name
+
+    # Ensure level is appropriate type.
+    level = Level.find_by_name(name)
```
Comment (laurelfan) [inline dashboard/app/dsl/level_group_dsl.rb:38]: This also ensures that levels exist, is this intentional? Probably a good idea... There's an argument for setting up the level group before the individual levels are done but they can leave them commented out or something if they want to do that.

File: dashboard/app/dsl/level_group_dsl.rb [hunk 2]
```diff
@@ -27,6 +28,19 @@ def page
   end
 
   def level(name)
+    # Ensure level name hasn't already been used.
+    if @level_names.include? name
```
Comment (laurelfan) [inline dashboard/app/dsl/level_group_dsl.rb:32]: Couldn't you just use @levels here?

**Comment by laurelfan** [comment]
LGTM once tests are fixed, looks like your change found a "real" problem with your test level ;)

File: dashboard/app/dsl/level_group_dsl.rb (hunk 2 above)
Comment (breville) [inline dashboard/app/dsl/level_group_dsl.rb:32]: @levels is cleared upon each page, but we want to make sure levels are unique for the entire level group.

Comment (breville) [inline dashboard/app/dsl/level_group_dsl.rb:32]: That said, I'm going to improve the names here.

File: dashboard/app/dsl/level_group_dsl.rb (hunk 1 above)
Comment (breville) [inline dashboard/app/dsl/level_group_dsl.rb:38]: Yes, I think this is a reasonable expectation and lets us do stronger validation.

**Comment by breville** [comment]
Hah, I fixed the test, and also added a couple more tests, which makes me happy.  This unit testing thing might really catch on!

**Pull request merged**